server/src/main/java/org/elasticsearch/index/engine/Engine.java

```diff
@@ -817,6 +817,12 @@ public final boolean refreshNeeded() {
     // NOTE: do NOT rename this to something containing flush or refresh!
     public abstract void writeIndexingBuffer() throws EngineException;
 
+    /**
+     * Checks if this engine should be flushed.
```

Comment:
can you explain that this can return false even if there are uncommitted changes. It's more of a maintainance function. maybe we should call it differently something like `shouldFlushForMaintainance` or `maintainanceFlushPending()` just suggestions to make it more clear

Reply:
Yannick and I came up with `shouldFlushToFreeTranslog`